cluster-autoscaler/utils/backoff/exponential_backoff.go

```diff
@@ -70,14 +70,12 @@ func (b *exponentialBackoff) Backoff(nodeGroup cloudprovider.NodeGroup, nodeInfo
 	duration := b.initialBackoffDuration
 	key := b.nodeGroupKey(nodeGroup)
 	if backoffInfo, found := b.backoffInfo[key]; found {
-		// Multiple concurrent scale-ups failing shouldn't cause backoff
-		// duration to increase, so we only increase it if we're not in
-		// backoff right now.
-		if backoffInfo.backoffUntil.Before(currentTime) {
-			duration = 2 * backoffInfo.duration
-			if duration > b.maxBackoffDuration {
-				duration = b.maxBackoffDuration
-			}
+		// Multiple concurrent scale-ups failing should cause backoff
+		// duration to increase, to not block cluster scale-down
+		// Increase backoff duration if we are in backoff already or not
+		duration = 2 * backoffInfo.duration
```

Comment:
I'd prefer to keep the if statement and just set duration correctly in else or something like that. 
A single failed scale-up can easily trigger calling Backoff() more than once (if you add multiple instances in one go it's likely for errors to crop up over multiple CA loop). With your change you're likely to just immediately jump to maximum backoff duration, rather than increase it exponentially as you retry.

Reply:
Ok that makes sense. I'll refactor it to keep the existing if statement and add logic to the else condition instead. Thanks for the additional context.

Reply:
@MaciekPytel I have updated it to something that more closely resembles what you're looking for but I'm having some questions. I know we want to increase the duration in `Backoff()` if the NG is already in backoff, but you're saying we do not want to do it on every reentrant call to `Backoff()` because then it goes up to the max too fast (like in my initial commit). Seems like we need some additional state in `backoffInfo` to inform how many retries we have, but you mention the solution should be only one LOC. Maybe you could provide the condition you're looking for?

Reply:
Right, so this requires more context than I initially thought. When NG goes off backoff, the backoff entry stays around for a few more hours (NodeGroupBackoffResetTimeout const). My idea when first implementing backoff was not to increase duration of the backoff if the NG is already in backoff (hence the if statement). The duration would be increased only if NG is not in backoff now, but has been in backoff recently (ie. the backoff entry is still there).

The rationale is that CA will not try another scale-up while the NG is already in backoff, so any new failure can likely be attributed to the original scale-up. Once NG goes off backoff and another scale-up fails the backoff duration will increase.

Not sure if this is the best way to do this, but your current approach doesn't fix the original issue - it is possible to set `duration = initialBackoffDuration` even if `found = true`.
I think the fix may be just as simple as adding `duration = backoffInfo.duration` in https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/utils/backoff/exponential_backoff.go#L75.

Reply:
Ok thanks for the context, this all makes more sense now. I will implement the approach and update the test.

Reply:
@MaciekPytel updated